packages/itwin/tree-widget/src/components/trees/stateless/models-tree/ModelsTree.tsx

```diff
@@ -94,35 +99,48 @@ export function StatelessModelsTree({
   );
 
   const getFocusedFilteredPaths = useMemo<GetFilteredPathsCallback | undefined>(() => {
+    isInstanceFocusError(errorRef.current) && setError(undefined);
     if (!loadFocusedInstancesKeys) {
       return undefined;
     }
     return async ({ imodelAccess }) => {
-      const targetKeys = await collectTargetKeys(loadFocusedInstancesKeys);
-
-      return ModelsTreeDefinition.createInstanceKeyPaths({
-        imodelAccess,
-        idsCache: getModelsTreeIdsCache(),
-        keys: targetKeys,
-        hierarchyConfig: hierarchyConfiguration,
-      });
+      try {
+        const targetKeys = await collectTargetKeys(loadFocusedInstancesKeys);
+        return await ModelsTreeDefinition.createInstanceKeyPaths({
+          imodelAccess,
+          idsCache: getModelsTreeIdsCache(),
+          keys: targetKeys,
+          hierarchyConfig: hierarchyConfiguration,
+        });
+      } catch (e) {
+        const newError = e instanceof Error && e.message.match(/Filter matches more than \d+ items/) ? "tooManyInstancesFocused" : "unknownInstanceFocusError";
+        setError(newError);
+        return [];
+      }
     };
-  }, [loadFocusedInstancesKeys, getModelsTreeIdsCache, hierarchyConfiguration]);
+  }, [loadFocusedInstancesKeys, getModelsTreeIdsCache, hierarchyConfiguration, errorRef]);
 
   const getSearchFilteredPaths = useMemo<GetFilteredPathsCallback | undefined>(() => {
+    isFilterError(errorRef.current) && setError(undefined);
```

Comment:
```suggestion
    setError(undefined);
```

This error is specific to filtering and only one kind of filtering can be applied (by instances or by label). You can always reset error when paths are reloaded

Reply:
You are right, changed to always clear the error and renamed to `filteringError` as it is more clear.